memory/transparent_hugepages.py

```diff
@@ -101,11 +101,14 @@ def test(self):
                                        ("thp_collapse_alloc"))
 
         # Check whether THP is Used or not
-        if thp_alloted_after <= thp_alloted_before:
+        if thp_alloted_after < thp_alloted_before:
             e_msg = "Thp usage count has not increased\n"
             e_msg += "Before Stress:%d\nAfter stress:%d" % (thp_alloted_before,
                                                             thp_alloted_after)
             self.fail(e_msg)
```

Comment:
Sorry for the late review. How is that `thp_alloted_after < thp_alloted_before` is an error condition whereas `thp_alloted_after == thp_alloted_before` is not ? As per the test condition, if we do not increase allocation count there might be one of the two reasons:

1.  System misconfiguration and we need to bailout
2.  Issue in the THP code

How are you checking for the second case, where you know that the system can allocate memory but failed to do so ?